projects/tinkerbell/hub/patches/0001-Add-Bootconfig-support-to-WriteFile-action.patch

```diff
@@ -0,0 +1,123 @@
+ golang.org/x/sys v0.0.0-20210124154548-22da62e12c0c h1:VwygUrnw9jn88c4u8GD3rZQbqrP/tgas88tPUbBxQrk=
+ golang.org/x/sys v0.0.0-20210124154548-22da62e12c0c/go.mod h1:h1NjWce9XRLGQEsW7wpKNCjG9DtNlClVuFLEZdDNbEs=
+-gopkg.in/check.v1 v0.0.0-20161208181325-20d25e280405 h1:yhCVgyC4o1eVCa2tZl7eS0r+SDo693bJlVdllGtEeKM=
+ gopkg.in/check.v1 v0.0.0-20161208181325-20d25e280405/go.mod h1:Co6ibVJAznAaIkqp8huTwlJQCZ016jof/cbN4VW5Yz0=
+ gopkg.in/check.v1 v1.0.0-20201130134442-10cb98267c6c h1:Hei/4ADfdWqJk1ZMxUNpqntNwaWcugrBjAiHlqqRiVk=
+ gopkg.in/check.v1 v1.0.0-20201130134442-10cb98267c6c/go.mod h1:JHkPIbrfpd72SG/EVd6muEfDQjcINNoR0C8j2r3qZ4Q=
+-gopkg.in/yaml.v3 v3.0.0-20200313102051-9f266ea9e77c h1:dUUwHk2QECo/6vqA44rthZ8ie2QXMNeKRTHCNY2nXvo=
+ gopkg.in/yaml.v3 v3.0.0-20200313102051-9f266ea9e77c/go.mod h1:K4uyk7z7BCEPqu6E+C64Yfv1cQ7kz7rIZviUmN+EgEM=
+ gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b h1:h8qDotaEPuJATrMmW04NCwg7v22aHH28wwpauUhK9Oo=
+ gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b/go.mod h1:K4uyk7z7BCEPqu6E+C64Yfv1cQ7kz7rIZviUmN+EgEM=
+diff --git a/actions/writefile/v1/main.go b/actions/writefile/v1/main.go
+index e5ed3cf..b38ef1c 100644
+--- a/actions/writefile/v1/main.go
++++ b/actions/writefile/v1/main.go
+@@ -5,6 +5,7 @@ import (
+ 	"fmt"
+ 	"io/ioutil"
+ 	"os"
++	"os/exec"
+ 	"path/filepath"
+ 	"strconv"
+ 	"strings"
+@@ -13,7 +14,10 @@ import (
+ 	log "github.com/sirupsen/logrus"
+ )
+ 
+-const mountAction = "/mountAction"
++const (
++	mountAction = "/mountAction"
++	bootConfigAction = "/bootconfig"
++)
+ 
+ func main() {
+ 	fmt.Printf("WriteFile - Write file to disk\n------------------------\n")
+@@ -23,6 +27,7 @@ func main() {
+ 	filePath := os.Getenv("DEST_PATH")
+ 
+ 	contents := os.Getenv("CONTENTS")
++	bootconfig := os.Getenv("BOOTCONFIG_CONTENTS")
+ 	uid := os.Getenv("UID")
+ 	gid := os.Getenv("GID")
+ 	mode := os.Getenv("MODE")
+@@ -42,6 +47,10 @@ func main() {
+ 		log.Fatalf("Could not parse mode: %v", err)
+ 	}
+ 
++	if bootconfig != "" && contents != "" {
++		log.Fatal("Both BOOTCONFIG_CONTENTS and CONTENTS cannot be set.")
++	}
++
+ 	fileMode := os.FileMode(modePrime)
+ 
+ 	dirModePrime, err := strconv.ParseUint(dirMode, 8, 32)
+@@ -82,12 +91,31 @@ func main() {
+ 		log.Fatalf("Failed to ensure directory exists: %v", err)
+ 	}
+ 
++	// If bootconfig is set, contents will be empty and will serve as output initrd file provided
++	// to bootconfig tool
+ 	fqFilePath := filepath.Join(mountAction, filePath)
+ 	// Write the file to disk
+ 	if err := ioutil.WriteFile(fqFilePath, []byte(contents), fileMode); err != nil {
+ 		log.Fatalf("Could not write file %s: %v", filePath, err)
+ 	}
+ 
++	if bootconfig != "" {
++		// Write the input bootconfig to file to serve as input to the tool
++		inputFilePath := "/userInputBootConfig"
++		err := ioutil.WriteFile(inputFilePath, []byte(bootconfig), fileMode)
++		if err != nil {
++			log.Fatalf("Could not write file %s: %v", filePath, err)
++		}
++		os.Remove(inputFilePath)
```

Comment:
Are we writing and immediately removing the file?

Reply:
good call, thats not supposed to be there.